vue-tsc Cannot find type definition file for 'vue/dist/__globaltypes_3.4_false.d.ts'.

### Vue - Official extension or vue-tsc version

2.1.2

### VSCode version

1.92.2

### Vue version

3.4.38

### TypeScript version

5.4.5

### System Info

_No response_

### Steps to reproduce

1. Create an empty project with `pnpm create vue@latest` using `typescript` and install dependencies.
![image](https://github.com/user-attachments/assets/01e4f377-0c03-44ca-8c0a-c82608af7638)

2. Run `pnpm type-check`  which runs `vue-tsc --build --force`
![image](https://github.com/user-attachments/assets/417023e7-fb52-40ce-a649-5e49dd1fcf8b)



### What is expected?

Get an output without errors

### What is actually happening?

The type-check script is failing with the following errors:
```
src/components/HelloWorld.vue:42:23 - error TS2688: Cannot find type definition file for 'vue/dist/__globaltypes_3.4_false.d.ts'.

42 /// <reference types="vue/dist/__globalTypes_3.4_false.d.ts" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/WelcomeItem.vue:88:23 - error TS2688: Cannot find type definition file for 'vue/dist/__globaltypes_3.4_false.d.ts'.

88 /// <reference types="vue/dist/__globalTypes_3.4_false.d.ts" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/icons/IconDocumentation.vue:8:23 - error TS2688: Cannot find type definition file for 'vue/dist/__globaltypes_3.4_false.d.ts'.

8 /// <reference types="vue/dist/__globalTypes_3.4_false.d.ts" />
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/icons/IconTooling.vue:20:23 - error TS2688: Cannot find type definition file for 'vue/dist/__globaltypes_3.4_false.d.ts'.

20 /// <reference types="vue/dist/__globalTypes_3.4_false.d.ts" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/icons/IconEcosystem.vue:8:23 - error TS2688: Cannot find type definition file for 'vue/dist/__globaltypes_3.4_false.d.ts'.

8 /// <reference types="vue/dist/__globalTypes_3.4_false.d.ts" />
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/icons/IconCommunity.vue:8:23 - error TS2688: Cannot find type definition file for 'vue/dist/__globaltypes_3.4_false.d.ts'.

8 /// <reference types="vue/dist/__globalTypes_3.4_false.d.ts" />
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/icons/IconSupport.vue:8:23 - error TS2688: Cannot find type definition file for 'vue/dist/__globaltypes_3.4_false.d.ts'.

8 /// <reference types="vue/dist/__globalTypes_3.4_false.d.ts" />
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/TheWelcome.vue:89:23 - error TS2688: Cannot find type definition file for 'vue/dist/__globaltypes_3.4_false.d.ts'.

89 /// <reference types="vue/dist/__globalTypes_3.4_false.d.ts" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/App.vue:48:23 - error TS2688: Cannot find type definition file for 'vue/dist/__globaltypes_3.4_false.d.ts'.

48 /// <reference types="vue/dist/__globalTypes_3.4_false.d.ts" />
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 9 errors.

 ELIFECYCLE  Command failed with exit code 2.

```

### Link to minimal reproduction

https://codesandbox.io/p/github/Tommytrg/vue-tsc-error/main?import=true

### Any additional comments?

_No response_

--- a/packages/component-meta/lib/base.ts
+++ b/packages/component-meta/lib/base.ts
@@ -137,7 +137,7 @@ export function baseCreate(
 
 	const fileExists = languageServiceHost.fileExists.bind(languageServiceHost);
 	const getScriptSnapshot = languageServiceHost.getScriptSnapshot.bind(languageServiceHost);
-	const globalTypesName = `__globalTypes_${commandLine.vueOptions.target}_${commandLine.vueOptions.strictTemplates}.d.ts`;
+	const globalTypesName = `__global_types_${commandLine.vueOptions.target}_${commandLine.vueOptions.strictTemplates}.d.ts`;
 	const snapshots = new Map<string, ts.IScriptSnapshot>();
 	languageServiceHost.fileExists = path => {
 		if (path.endsWith(globalTypesName)) {

--- a/packages/language-core/lib/codegen/script/index.ts
+++ b/packages/language-core/lib/codegen/script/index.ts
@@ -51,7 +51,7 @@ export interface ScriptCodegenOptions {
 export function* generateScript(options: ScriptCodegenOptions): Generator<Code, ScriptCodegenContext> {
 	const ctx = createScriptCodegenContext(options);
 
-	yield `/// <reference types="${options.vueCompilerOptions.lib}/dist/__globalTypes_${options.vueCompilerOptions.target}_${options.vueCompilerOptions.strictTemplates}.d.ts" />${newLine}`;
+	yield `/// <reference types="${options.vueCompilerOptions.lib}/dist/__global_types_${options.vueCompilerOptions.target}_${options.vueCompilerOptions.strictTemplates}.d.ts" />${newLine}`;
 
 	if (options.sfc.script?.src) {
 		yield* generateSrc(options.sfc.script, options.sfc.script.src);

--- a/packages/language-server/lib/initialize.ts
+++ b/packages/language-server/lib/initialize.ts
@@ -52,7 +52,7 @@ export function initialize(
 						project.vue = { compilerOptions: vueCompilerOptions };
 
 						if (project.typescript) {
-							const globalTypesName = `__globalTypes_${vueCompilerOptions.target}_${vueCompilerOptions.strictTemplates}.d.ts`;
+							const globalTypesName = `__global_types_${vueCompilerOptions.target}_${vueCompilerOptions.strictTemplates}.d.ts`;
 							const fileExists = project.typescript.languageServiceHost.fileExists.bind(project.typescript.languageServiceHost);
 							const getScriptSnapshot = project.typescript.languageServiceHost.getScriptSnapshot.bind(project.typescript.languageServiceHost);
 							const snapshots = new Map<string, ts.IScriptSnapshot>();

--- a/packages/tsc/index.ts
+++ b/packages/tsc/index.ts
@@ -27,7 +27,7 @@ export function run(tscPath = require.resolve('typescript/lib/tsc')) {
 						: options.host?.getCurrentDirectory() ?? ts.sys.getCurrentDirectory();
 					const libDir = require.resolve(`${vueOptions.lib}/package.json`, { paths: [rootDir] })
 						.slice(0, -'package.json'.length);
-					const globalTypesPath = `${libDir}dist/__globalTypes_${vueOptions.target}_${vueOptions.strictTemplates}.d.ts`;
+					const globalTypesPath = `${libDir}dist/__global_types_${vueOptions.target}_${vueOptions.strictTemplates}.d.ts`;
 					const globalTypesContents = vue.generateGlobalTypes(vueOptions.lib, vueOptions.target, vueOptions.strictTemplates);
 					ts.sys.writeFile(globalTypesPath, globalTypesContents);
 				} catch { }

--- a/packages/tsc/tests/dts.spec.ts
+++ b/packages/tsc/tests/dts.spec.ts
@@ -37,7 +37,7 @@ describe('vue-tsc-dts', () => {
 				: options.host?.getCurrentDirectory() ?? ts.sys.getCurrentDirectory();
 			const libDir = require.resolve(`${vueOptions.lib}/package.json`, { paths: [rootDir] })
 				.slice(0, -'package.json'.length);
-			const globalTypesPath = `${libDir}dist/__globalTypes_${vueOptions.target}_${vueOptions.strictTemplates}.d.ts`;
+			const globalTypesPath = `${libDir}dist/__global_types_${vueOptions.target}_${vueOptions.strictTemplates}.d.ts`;
 			const globalTypesContents = vue.generateGlobalTypes(vueOptions.lib, vueOptions.target, vueOptions.strictTemplates);
 			ts.sys.writeFile(globalTypesPath, globalTypesContents);
 		} catch { }

--- a/packages/typescript-plugin/index.ts
+++ b/packages/typescript-plugin/index.ts
@@ -63,7 +63,7 @@ const plugin: ts.server.PluginModuleFactory = mods => {
 				try {
 					const libDir = require.resolve(`${options.lib}/package.json`, { paths: [proj.getCurrentDirectory()] })
 						.slice(0, -'package.json'.length);
-					const globalTypesPath = `${libDir}dist/__globalTypes_${options.target}_${options.strictTemplates}.d.ts`;
+					const globalTypesPath = `${libDir}dist/__global_types_${options.target}_${options.strictTemplates}.d.ts`;
 					const globalTypesContents = vue.generateGlobalTypes(options.lib, options.target, options.strictTemplates);
 					proj.writeFile(globalTypesPath, globalTypesContents);
 				} catch { }

